### PR TITLE
Fix core dump when compile expression

### PR DESCRIPTION
**Pull request by liuneng1994**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix core dump when compile expression in some case

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by liuneng1994** [comment]
reproduce: https://fiddle.clickhouse.com/b945bb64-a230-49ab-acdd-099cf7348255

**Comment by liuneng1994** [comment]
This bug is caused by PR https://github.com/ClickHouse/ClickHouse/pull/50531

**Comment by liuneng1994** [comment]
@kitaisreal Please review

**Review by alexey-milovidov — APPROVED** [review]

**Comment by kitaisreal** [comment]
@liuneng1994 why it does not crashed before ?

File: src/DataTypes/Native.cpp [hunk 1]
```diff
@@ -123,9 +123,11 @@ llvm::Value * nativeCast(llvm::IRBuilderBase & b, const DataTypePtr & from_type,
     }
     else if (to_type->isNullable())
     {
-        auto * from_native_type = toNativeType(b, from_type);
+        auto * to_native_type = toNativeType(b, to_type);
         auto * inner = nativeCast(b, from_type, value, removeNullable(to_type));
-        return b.CreateInsertValue(llvm::Constant::getNullValue(from_native_type), inner, {0});
+        auto * res_ptr = b.CreateAlloca(to_native_type);
```
Comment (kitaisreal) [inline src/DataTypes/Native.cpp:128]: I am not sure if after Alloca, NullValue is allocated, and you can load and insert into it.

**Review by kitaisreal — CHANGES_REQUESTED** [review]

**Comment by liuneng1994** [comment]
> @liuneng1994 why it does not crashed before ?



> @liuneng1994 why it does not crashed before ?

When from type is Float, to type is Nullable(Float), CreateInsertValue will failed, because getNullValue will return Float Value not a struct type value

**Comment by baibaichen** [comment]
> @liuneng1994 why it does not crashed before ?

@kitaisreal caused by new intrduced file native.cpp

File: src/DataTypes/Native.cpp (hunk 1 above)
Comment (liuneng1994) [reply src/DataTypes/Native.cpp]: You are right, Alloca just allocate memory without initialization.

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**